src/main/java/org/snomed/snowstorm/core/data/services/ConceptService.java

```diff
@@ -153,55 +158,62 @@ public Concept find(String id, List<LanguageDialect> languageDialects, BranchTim
 	}
 
 	public ConceptHistory loadConceptHistory(String conceptId, List<CodeSystemVersion> codeSystemVersions, boolean showFutureVersions) {
+		int now = Integer.parseInt(DATE_TIME_FORMATTER.format(Instant.now()));
 		Map<String, BranchCriteria> branchCriteria = new HashMap<>();
 		for (CodeSystemVersion codeSystemVersion : codeSystemVersions) {
-			String branchPath = codeSystemVersion.getBranchPath();
-			branchCriteria.put(branchPath, versionControlHelper.getBranchCriteria(branchPath));
+			if (showFutureVersions || codeSystemVersion.getEffectiveDate() < now) {
+				String branchPath = codeSystemVersion.getBranchPath();
+				branchCriteria.put(branchPath, versionControlHelper.getBranchCriteria(branchPath));
+			}
 		}
 
-		Function<String, BoolQueryBuilder> defaultBoolQueryFunction = cId -> {
-			BoolQueryBuilder someReleaseBranch = boolQuery();
-			BoolQueryBuilder boolQueryBuilder = boolQuery();
+		BiFunction<String, ComponentType, BoolQueryBuilder> defaultFullQuery = (cId, cT) -> {
```

Comment:
I have renamed the variables associated with `boolQuery` to help readability and nested where possible.

Reply:
Looks good. I find it easier to read when a separate boolQuery is used to hold the should clauses rather than holding should and must clauses against the same boolQuery but the `minimumShouldMatch` method seems to work. This is fine.

Reply:
Ah, I also had that idea and originally had that code as it was "grouped" better. I figured it was OTT, but good to know going forward!